src/main/java/org/apache/sling/event/impl/jobs/stats/GaugeSupport.java

```diff
@@ -150,9 +154,10 @@ private void registerWithSuffix(String suffix, int count, Gauge<Long> value) {
             metricRegistry.register(metricName, value);
             gaugeMetricNames.add(metricName);
         } catch (IllegalArgumentException e) {
-            if (queueName != null) {
-                registerWithSuffix(suffix, count + 1, value);
-            }
+            logger.error("Failed to register mbean with suffix " + suffix, e);
+            // if (queueName != null) {
+            //     registerWithSuffix(suffix, count + 1, value);
+            // }
```

Comment:
Rather than just failing, this should add a recursion limit. I wouldn't _think_ it would have to be large say 5(?), but it should retry x times incrementing the count and then log the error and quit.

Reply:
+1 to what @klcodanr recommends.

This should do a log.debug for the first `n` times, then when it hits the limit `n` only do a log.error. The way it is done now is removing support for metric name collisions due to name sanitization (which is one legitimate reason for this retry)

> I wouldn't think it would have to be large say 5(?)

agree, I thought about `10` initially, but could be somewhere in that range, yes.